Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -0,0 +1,388 @@
+                matlabbatch{1}.spm.spatial.realign.write.roptions.which     = [2 0];
+                matlabbatch{1}.spm.spatial.realign.write.roptions.interp    = 4;
+                matlabbatch{1}.spm.spatial.realign.write.roptions.wrap      = [0 0 0];
+                matlabbatch{1}.spm.spatial.realign.write.roptions.mask      = 1;
+                matlabbatch{1}.spm.spatial.realign.write.roptions.prefix    = 'r';
+            
+                spm_jobman('run',matlabbatch); % this applies the motion correction in native space
+                
+                path_rASL4D = fullfile(Fpath, ['r' Ffile '.nii']);
+                
+                % RELOAD ASL4D.nii
+                % Note that we only reload ASL4D.nii here if we have motion corrected timeseries.
+                % If path_ASL4D was not a path, or if sideCarMat didn't exist (motion estimation was not performed)
+                % or if nVolumes==1 (we did not have multiple ASL volumes) there is no reason to resample and we keep using
+                % the already above loaded ASL4D.nii
+                ASL4D = xASL_io_Nifti2Im(path_rASL4D);
+                xASL_delete(path_rASL4D);
+            end
+        end
+    end
+
+
+    %% 2B. Apply time decoding if needed
+    if x.modules.asl.bTimeEncoded
+        %% #997 Check that the motion correction of Hadamard is correctly performed here,
+        % should we use path_rASL4D from here onwards, which defaults to path_ASL4D?
+
+        % Decoding of TimeEncoded data - it outputs the decoded image as well as the parameters
+        [PWI4D, Control4D, x.Q] = xASL_quant_HadamardDecoding(path_ASL4D, x.Q);
+	    nVolumes = size(PWI4D, 4);
+
+        bCreatePWI3D = true; % because we can do this now
+        bCreateControl3D = true; % because we can do this now        
+
+
+    %% 2C. Non-time encoded subtraction
+    else
+        if nVolumes>1 && ~x.modules.asl.bContainsDeltaM
+            % Paired subtraction
+            [Control4D, Label4D] = xASL_quant_GetControlLabelOrder(ASL4D);
+            PWI4D = Control4D - Label4D;
+            
+            % Skip every other value in the vectors as they were stored for both control and label images 
+            x.Q.EchoTime_PWI4D = x.Q.EchoTime(1:2:end);
+            x.Q.InitialPLD_PWI4D = x.Q.Initial_PLD(1:2:end);
+            x.Q.LabelingDuration_PWI4D = x.Q.LabelingDuration(1:2:end);
+            % In this case, the vectors are equal for PWI4D and Control4D
+            x.Q.EchoTime_Control4D = x.Q.EchoTime_PWI4D;
+            x.Q.InitialPLD_Control4D = x.Q.InitialPLD_PWI4D;
+            x.Q.LabelingDuration_Control4D = x.Q.LabelingDuration_PWI4D;
+
+            bCreatePWI3D = true; % because we can do this now
+            bCreateControl3D = true; % because we can do this now
+        elseif ~x.modules.asl.bContainsDeltaM
+            warning('Single volume detected that was not subtracted, cannot create PWI images');
+            PWI4D = [];
+            Control4D = ASl4D;
```

Comment:
ASL4D not ASl4D

Reply:
Done in 2736b6bb8e3682493372bff9f20f0863e810b174